File: OneDimEnergy.py
import numpy as np

def OneDimEnergy(V, Psi, dz, enorm, U0, Sigma, w, TrapPotential):
    # Calculates the energy densityof the condensates

    Sig2 = Sigma*Sigma
    Density = Psi*np.conj(Psi)
    Psi_X = np.gradient(Psi)

    KE = 0.5*(Psi_X*np.conj(Psi_X))/(dz*dz)
    VE = TrapPotential*Density+(0.5/Sig2)+(0.5*w**2*Sig2)
    CE = (U0/2)*Density*Density

    EK = np.sum(KE)*dz/enorm
    EV = np.sum(VE)*dz/enorm
    EC = np.sum(CE)*dz/enorm

    return EK, EV, EC

File: OneDimContact MatLab/OneDimEnergy.m
function [EK,EV,EC] = OneDimEnergy(V,Psi,dz,enorm,U0,Sigma,w,TrapPotential)

% This function calculates the energy density of the condensate

Sig2 = Sigma.*Sigma;
Density = Psi.*conj(Psi); 
[Psi_X] = gradient(Psi); 

KE = (1/2.0)*(Psi_X.*conj(Psi_X))/(dz*dz);                    %kinetic                 
VE = TrapPotential.*Density+(1/2.0)./Sig2+(1/2.0)*w^2*Sig2;   % trap and mf
CE = (U0/2.0)*Density.*Density;                               % contact

EK = sum(KE)*dz/enorm;          % these sums mimic intergral over all space                    
EV = sum(VE)*dz/enorm;
EC = sum(CE)*dz/enorm;

end

